apps/nft.blocksense.network/workers/socialDataAPI.ts

```diff
@@ -0,0 +1,60 @@
+'use server';
+
+const BLOCKSENSE_ID = process.env['NEXT_PUBLIC_BLOCKSENSE_X_ID'];
+const API_KEY = process.env['NEXT_PUBLIC_SOCIAL_DATA_API_KEY'];
+const TWEET_ID = '1913586022836785624'; // TODO: with the actual tweet ID
+
+export async function getUserIdByUsername(username: string) {
+  const response = await fetch(
```

Comment:
```suggestion
  // Remove the '@' from the xHandle
  const username = xHandle.slice(1);
  const response = await fetch(
```

Reply:
With that said we can update `onXHandleChange` as follows:
```ts
  const onXHandleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const value = e.target.value;
    setXHandle(value);

    // Regex to check if the input starts with '@'
    if (!/^@/.test(value)) {
        setXError('Incorrect X handle. It should start with "@"');
        setXSuccess('');
    } else {
        setXError('');
    }
};
```

Reply:
After an offline discussion, we decided it's best to allow users to set both `@hanele` and `handle` here